src/js/solo/goldify-playlist/GoldifyPlaylistData.js

```diff
@@ -29,61 +51,204 @@ class GoldifyPlaylistData extends Component {
         if (data === undefined || data.error) {
           replaceWindowURL("/");
         } else {
-          this.setState({
-            goldifyPlaylistData: data,
-          });
+          this.setInitialGoldifyPlaylistData(data);
         }
       }
     );
   }
 
+  setInitialGoldifyPlaylistData(playlistTrackData) {
+    this.setSavedGoldifyPlaylistData(playlistTrackData);
+    var uriList = [];
+    for (var i = 0; i < playlistTrackData.items.length; ++i) {
+      var track = playlistTrackData.items[i].track;
+      uriList.push(track.uri);
+    }
+    this.goldifyPlaylistTrackUriList = uriList;
+    this.setState({
+      goldifyPlaylistData: playlistTrackData,
+      playlistDirty: false,
+    });
+  }
+
+  setSavedGoldifyPlaylistData = (data) => {
+    this.savedGoldifyPlaylistData = JSON.parse(JSON.stringify(data));
+  };
+
+  addGoldifyPlaylistTrackUri = (trackUri) => {
+    this.goldifyPlaylistTrackUriList.push(trackUri);
+  };
+
+  removeGoldifyPlaylistTrackUri = (trackUri) => {
+    var index = this.goldifyPlaylistTrackUriList.indexOf(trackUri);
+    if (index !== -1) {
+      this.goldifyPlaylistTrackUriList.splice(index, 1);
+    } else {
+      throw Error("Unable to remove track: " + trackUri);
+    }
+  };
+
+  updateGoldifyPlaylist() {
+    if (this.goldifyPlaylistTrackUriList !== undefined) {
+      replacePlaylistTracks(
+        this.props.retrievedTokenData,
+        this.props.goldifyPlaylistId,
+        this.goldifyPlaylistTrackUriList
+      );
+      this.setSavedGoldifyPlaylistData(this.state.goldifyPlaylistData);
+      this.setState({
+        playlistDirty: false,
+      });
+    } else {
+      throw Error("Playlist Track Uri List cannot be undefined.");
```

Comment:
Yea that's a good point, I think I threw that in just as a safety precaution but I don't think this could occur. Removing in a commit then I'll push it